### DISCUSSION TITLE
I built a Minidexed in Eurorack form factor for my entirely DIY modular

### DISCUSSION THREAD

**Discussion by AriaSalvatrice** [probonopd/MiniDexed#524] [Show and tell]
Heyo, just wanted to show off my contraption

![IMG_20230728_083637b](https://github.com/probonopd/MiniDexed/assets/25130242/19d0164c-561f-4949-bb2c-b1d176b69a15)

https://www.youtube.com/watch?v=7ot3kDXmUb0

It takes power from the +12V rail, so I used a cheapo Aliexpress buck converter module, the project used too much current for a simple regulator. 

As of now, it only has MIDI in to Audio out, I expect to give it a little CV to MIDI expander soon. It's using a RPI zero, which suits me just fine since once I rig up CV input I will use it mostly monophonically. 

I've posted a little build thread on Fedi if anyone's interested (linking you to the last current reply in the thread): https://fedi.aria.dog/notice/AY998Ac7h8OVgQfHai

Thanks for putting together this project!

**Comment by probonopd** [comment]
Very cool @AriaSalvatrice. Thanks for posting, and have fun with MiniDexed!

Hopefully the RPi Zero 2 becomes available soon again. It's 8x better for MiniDexed than the Pi Zero ;-)

**Comment by DutchBigEd** [comment]
Nice built and a good idea to use a buck converter! I made an eurorack module myself  and noticed the 7805 gets hot.
Also I read on your webpage about your concern about the width of the display so I used a 0.9" oled display, made my module 30mm wide.

**Comment by BobanSpasic** [comment]
There is a 7805 in TO-3 packaging/case. Much better cooling possibilities.

**Reply by Gustavomurta** [reply]
I do not recommend the use of the 7805 series regulator in this type of project due to the low efficiency.
I used the LM2596 regulator module set to 5V on the output. You don't even need a heatsink.
The power supply can be between 12V and 9V (most suitable).

**Reply by AriaSalvatrice** [reply]
Even if current used were no concern, I found that a 7805 with a heatsink would still overheat, causing screen flickers and unstable operation (see that little IR thermometer in the video? It reached 80°C).

**Comment by AriaSalvatrice** [comment]
I've been working on taking it further, to accept CV input, and after a few experiments with a protoboard, decided to make a full PCB for a Minidexed modular conversion.

I've been posting about my idea & WIP plans here, it might be of interest, even if CV input is considered out of scope: 

https://lookmumnocomputer.discourse.group/t/working-on-a-minidexed-modular-conversion/6433

**Reply by probonopd** [reply]
Very cool @AriaSalvatrice. CV is out of scope for the MiniDexed project itself, as the DX line of instruments are purely MIDI, but doing this in an add-on project is the way to go. Great work, keep it up! :+1:

**Reply by DutchBigEd** [reply]
I’ve read you article at LMNC, and I noticed you use 3.3 volt @ R1 in the midi out schematic. Do you have a specific reason why not using 5 volt there?

**Reply by AriaSalvatrice** [reply]
Well, flipping the question around, do I have a specific reason not to use 3.3V? 

The MIDI specification is quite an impenetrable tome I don't understand much, but as far as I know, it recommends using 3.3V for new builds since the 2014 1.1 revision, and it states it should remain compatible with 1.0 devices with a properly implemented MIDI IN circuit with an opto-isolator. Any reason I should disregard this spec?

**Reply by DutchBigEd** [reply]
A picture says more than a thousand words…
Source: https://www.midi.org/specifications-old/item/midi-din-electrical-specification
<img width="833" alt="5F8C3F40-9FD9-44B4-A32D-01908F2A2E4F" src="https://github.com/probonopd/MiniDexed/assets/17274454/cc76f3f2-6134-4883-813c-aaa00b2e2fcd">

**Reply by AriaSalvatrice** [reply]
ah right, i thought it was the choice of voltage that was contentious here, rather than it having mismatched resistors

**Reply by DutchBigEd** [reply]
My choise would be to use 5 volt with the 220 ohm resistors for several reasons.
1. Arduino is powered at 5 volt so no need for a buffer.
2. The error margin is bigger: 10% @ 5 volt or 5% @ 3.3 volt.
3. You can exchange the voltage converter by two resistors (2k and 3k3)

**Reply by diyelectromusic** [reply]
As I understand it, the key thing is to ensure the 5mA MIDI current loop is working properly and has clean on and off conditions.  The resistor values given in the spec ensure that in the case for the two different supply voltages.

But the key thing here is that the voltage source matches the logic level of the output device as when the loop is complete (i.e. it is plugged into something with its own optoisolator at the other end) a pulled high TX pin on MIDI pin 5 will need to match the VCC level on MIDI pin 4.  If VCC is lower then presumably a current will flow which may or may not erroneously activate the optoisolator, depending on its spec I guess, when it should be remaining inactive.  I guess when TX is pulled low a current can flow from either 3V3 or 5V so that part is probably ok.

I'm not an electronics person so might not have got the details exactly right, but I'm pretty sure a 5V pulled up TX shouldn't connect to a VCC at 3V3...

If you've not already seen it, the notesandvolts explanation of how MIDI works is excellent!  See https://www.youtube.com/watch?v=0L7WAMFWSgY

Kevin

**Reply by AriaSalvatrice** [reply]
The mismatched voltages on the MIDI out were absolutely an oversight, but it was fixed in time for the board sent to the fab! I intended to go with 3V3 as in the 2014 MIDI spec, but I realized that in addition to the advantages @DutchBigEd mentioned, going with 5V meant a simpler BOM, so I wouldn't have to specify an additional resistor type.

**Reply by diyelectromusic** [reply]
Great that was fixed before fab :)

But for completeness, the MIDI spec was updated* to support either 5V or 3V3 not for 3V3 to replace 5V.

For the reasons I suggest, VCC has to match the voltage level of the TX signal or when you plug in a MIDI lead that completes the circuit (by linking your device to the receivers optoisolator) you'll end up with 5V connected to 3V3 via the remote circuit.

So it isn't that using 5V is advantageous, it is that if you don't use the same VCC as the logic signaling then at some point the 5V TX will meet the 3V3 VCC which I'm sure wouldn't end well - especially if the 3V3 was originally coming from the Pi GPIO - I don't know what it would think of trying to sink 5V on its 3V3 power pins, but it sounds expensive to me :)

Kevin
* Spec Version: "(CA-033) MIDI 1.0 Electrical Specification Update [2014]"

**Comment by AriaSalvatrice** [comment]
Revision 1 is off to print! If/when it works I'll clean up the files for release, and write up build instructions.

I'm calling the project as a whole "E.PIANO 1" to make it clear it's its own thing that bundles a Minidexed core rather than an official offshoot of the MiniDexed project. 

I've posted the revised schematic here: https://lookmumnocomputer.discourse.group/t/working-on-a-minidexed-modular-conversion/6433/6

**Comment by AriaSalvatrice** [comment]
The boards arrived, and they passed the smoke test!

![fb7ecfd35da7e1ce19bb0943cd5830db6d7bdd11096179e3e22ca6b8e7500806](https://github.com/probonopd/MiniDexed/assets/25130242/ae426392-7a3c-45eb-8ecb-cc3999a7e548)
![e61b192472e7c64ee1abd0c2d9e15cb092244a8a5d2f4a0daacbc62545f4de43](https://github.com/probonopd/MiniDexed/assets/25130242/65e6c25b-6d36-4c6f-b222-280905226295)
![a03b8570033e5e38827f67f8440dd03a09107f5b429bcba85d4668d3480cb70c](https://github.com/probonopd/MiniDexed/assets/25130242/fffd1fc8-d9e3-4813-ae85-9be4425e92b4)
![ac4a3e05714104762951d098aa789e307213edf4c7f4e4c01e433a7bc7053e2a](https://github.com/probonopd/MiniDexed/assets/25130242/7cc1000e-5b33-40cd-aa94-09942c3bdf1a)

Here you can see how it will fit in Kosmo size. (20cm tall DIY format with big jacks). I intend to also make a panel design for Eurorack size, maybe PCB material. It will require a lot of panel wiring, not a build for the faint of heart, lol.

Still a lot of work left, but it’s great news it all fits mechanically without smoking out. 

I still need to write the Arduino MIDI<>CV C++ firmware, it will take a while to tweak it to be just right, but it’s a pretty easy task

**Reply by AriaSalvatrice** [reply]
A little update: after having trouble with a bad chip and ordering new stock, I think I have the electronics part 100% working. Now it's just a matter of making the firmware! Thee Arduino is working as a minimal MIDI Thru here. Just need to make the rest of the I/O useful

Video demo: https://makertube.net/w/7962de80-b94d-47b6-b344-530da7696a7a
Alternate icky ad-laden youtube link: https://www.youtube.com/watch?v=J280357UB_U

I think the PCB I designed has no issues, but I might revise it to make a Eurorack module that's easier to wire up, with stacked boards.

**Comment by Saddox** [comment]
Nice work! I would really like a Eurorack-remix on this :)

**Reply by AriaSalvatrice** [reply]
I didn't have much time to work on it since I last posted but that could be the plan depending on interest! It's electrically 100% compatible but it needs a LOT of panel wiring as it stands. I developed my own techniques & software centered around 3D printing and wire wrap so I tend to be able to stomach a lot more panel wiring than most people lol. 

Once I get stuff working it might be worth designing it as a stackup of PCBs usable as a real module with a PCB faceplate and no/minimal wiring.

**Reply by Saddox** [reply]
Yes, it would be worth a Eurorack-panel :) 

I myself do not mind wiring but less wiring is good.

**Comment by AriaSalvatrice** [comment]
Took a while, but with the help of @diyelectromusic, I added to the MiniDexed a minimal facility to display messages from MIDI Sysex, so I can send text messages from the Arduino, creating the illusion of a single cohesive device for the end-user. As soon as you use thee big rotary encoder, the MiniDexed takes over the display again.

![218756a4ede999fe21457d1079e561d6eb1e933939bfe5af31c740f398ac7162](https://github.com/probonopd/MiniDexed/assets/25130242/8fcdeac8-69ef-415f-933e-df2a4d49a2e2)

You can see how I went about it here (also had to make the `LCDWrite` function public in the corresponding header) - it's a very minimal hack focusing exclusively on my own needs: 

https://github.com/AriaSalvatrice/MiniDexed/blob/4e809b7688a9c657f6914d236d47404ad61c7117/src/mididevice.cpp#L184
